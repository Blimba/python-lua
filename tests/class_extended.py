class example:
    def __init__(self, v):
        self.value = v

    def __add__(self, other):
        return example(self.value+other.value)

    def __sub__(self, other):
        return example(self.value-other.value)

    def __mul__(self, other):
        return example(self.value*other.value)

    def __truediv__(self, other):
        return example(self.value/other.value)

    def __lt__(self, other):
        return self.value < other.value

    def __str__(self):
        return str(self.value)

a = example(5)
b = example(6)
print(a+b)
print(a-b)
print(a*b)
print(a/b)
