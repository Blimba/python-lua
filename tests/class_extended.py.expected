11
-1
30
0.83333333333333
5
5
6
